README.md

```diff
@@ -0,0 +1,190 @@
+# Charmed Data Integrator Operator
+
+## Overview
+
+This charm allows to retrieve product credentials needed to authenticate (SCRAM) with different kinds of data platform products([MongoDB](https://github.com/canonical/mongodb-operator), [MySQL](https://github.com/canonical/mysql-operator), [PostgreSQL](https://github.com/canonical/postgresql-operator), and [Kafka](https://github.com/canonical/kafka-operator)). 
+
+
+## Config options
+
+The supported configuration options are the following: 
+
+database-name - `string`; The desired database name for which the access will be granted.
+
+topic-name - `string`; The topic name for which the access is granted.
+
+extra-user-roles - `string`; a comma-separated list of values that contains the required extra roles, e.g. `admin` in case of a database or `producer`, `consumer` in case of Kafka. 
+
+
+## Usage
+
+### Basic Usage
+To deploy a unit of the Data Integrator Charm.
+
+#### Charmhub
+```shell
+juju deploy data-integrator --channel edge
+```
+#### From source
+```shell
+git clone https://github.com/canonical/data-integrator.git
+cd data-integrator/
+sudo snap install charmcraft --classic
+charmcraft pack
+```
+Then,
+```shell
+juju deploy ./data-integrator_ubuntu-22.04-amd64.charm
+```
+
+#### Configuration
+
+If you want to require access to a database, specify the `database-name` parameter: 
+
+```shell
+juju config data-integrator database-name=test-database 
+```
+In addition, required `extra-user-roles` can be specified. 
+
+```shell
+juju config data-integrator database-name=test-database extra-user-roles=admin
+```
+
+Instead, for Kafka please configure the desired `topic-name`:
+
+```shell
+juju config data-integrator topic-name=test-topic extra-user-roles=producer,consumer
+```
+
+#### Relation with desidered application 
+
+In order to related to the desired data-platform application use the following command: 
+
+```shell
+juju relate data-integrator <application>
+```
+
+After the relation has been created, the credentials and connection information can be retrieved with an action.
+
+In order to change the current credentials (username and password), remove the relation with the application and establish a new one. 
+When the relation is removed, the access with the previous credentials will be removed.
+
+```shell
+juju remove-relation data-integrator <application>
+```
+
+If you need to modify `database-name`, `topic-name` or `extra-user-roles` and the relation has been already established, you need to remove the relation and then change the `database-name`, `topic-name` or `extra-user-roles`, and finally relate the data-integrator with the desidered application.
+
+#### Retrieve credentials
+
+In order to retrieve the credentials use the following action:
+
+```shell
+juju run-action data-integrator/leader get-credentials --wait
+```
+
+
+#### Relate Data Integrator with MongoDB
```

Comment:
Nice, I would also call this a "Tutorial". Maybe it is worth putting into a separated file called Tutorial

Reply:
I will add a section named "Tutorial" but I would like to keep it inside the README since it is a pretty minimal tutorial and the README is pretty short. Anyway I will trust your opinion on this point.